### ISSUE TITLE
Adaptive layout for wide screens

### ISSUE THREAD

**Issue by ShadowJonathan** [vector-im/element-web#18072]
(Breakout of https://github.com/matrix-org/matrix-react-sdk/pull/6291#pullrequestreview-707090023)

#### Is your suggestion related to a problem? Please describe.
When using bubble messages on a wide screen, it's often hard to follow along the conversation when i participate in it, as my messages are aligned to the far right of the screen.

#### Describe the solution you'd like.
Take after telegram. Telegram has an option called "Adaptive layout for wide screens", that aligns the messages to the same side after a certain chat window width;

![](https://user-images.githubusercontent.com/22740616/125755955-c01c6389-10cd-435a-8194-5bd2ca3759cc.png)

#### Additional context
Also see https://github.com/SchildiChat/schildichat-desktop/issues/33

**Comment by robintown** [comment]
Crazy idea but perhaps Element could automatically switch between modern layout for wide screens, and bubble layout for narrow screens?

**Comment by SimonBrandner** [comment]
> Crazy idea but perhaps Element could automatically transition between modern layout for wide screens, and bubble layout for narrow screens? (so it'd be an additional "Adaptive" layout in appearance settings)

That sounds interesting, though I think for that to work we would need the layouts to be more similar. E.g. currently replies look completely different (which I am not a big fan of btw) which would make this feel weird

**Comment by ShadowJonathan** [comment]
It depends, I'm inclined to say no because the different style might be confusing to the user, and maybe not something a current user would like to look at.

**Comment by amshakal** [comment]
I agree with the problem. On larger screens it's harder to follow along the conversation because the messages are too far apart. Maybe a simpler solution to begin with is the center align the messages from sender and reciever so that there is always an overlap between the two, making it easy to follow along, similar to how whatsapp does it. I'm adding a screenshot for reference.
<img width="648" alt="Screenshot 2021-07-21 at 12 07 45" src="https://user-images.githubusercontent.com/85162587/126479433-06c841f1-6ef6-4cef-9b45-56b65c51cf76.png">

**Comment by ShadowJonathan** [comment]
Re: that last image, I think it's also useful to narrow the composer to that width, to keep the composer-timeline "sync"

**Comment by amshakal** [comment]
Oh interesting, let me make a quick mock up and see how that looks. :)

**Comment by daniellekirkwood** [comment]
@amshakal do we want to go with a "max-width" approach and centre align the timeline when bubbles are on? (Just like your screenshot...)

**Comment by amshakal** [comment]
I think centered so it's easy parse through all the messages. (Just like the screenshot)

**Comment by amshakal** [comment]
To discuss: If composer and header should stay edge to edge or aligned to the timeline messages.

**Comment by ShadowJonathan** [comment]
To that; yes, currently on `develop.` it is already misaligned to the left edge, and it becomes distracting with the current room.

Discord has this when the channel disables file-sending, and i always noticed it when that happens. They're currently trying to smartly "polyfill" that by making the `+` button on the left do multiple things (like send commands, or stickers), but that doesn't fix the underlying weirdness i notice.

**Comment by t3chguy** [comment]
@amshakal what is the max-width value (in px) in your Figma there?

**Comment by amshakal** [comment]
I kept it as 1200px in the Figma file. Will test it once to make sure implemented.

**Comment by ShadowJonathan** [comment]
In the Element Design room today, there was someone asking for the original design proposal (In wide screen, align all messages (including the user's own) to the left of the screen.

Could this issue maybe be reopened? Or should a seperate issue be made to address this? (Considering the feature is now out of labs)
